pkg/agent/nodeportlocal/portcache/port_table.go

```diff
@@ -77,10 +88,85 @@ type PortTable struct {
 	tableLock        sync.RWMutex
 }
 
+func GetPortTableKey(obj interface{}) (string, error) {
+	npData := obj.(*CacheNpData)
+	var key string
+	if npData.Type == "NodePortTable" {
+		key = NodePortProtoFormat(npData.Data.NodePort, npData.Data.Protocol.Protocol)
+	} else {
+		// Use the podIPPort name as the key by default.
+		key = podIPPortProtoFormat(npData.Data.PodIP, npData.Data.PodPort, npData.Data.Protocol.Protocol)
+	}
+	return key, nil
+}
+
+func (pt *PortTable) addKeyFromPortTableCache(cacheType string, key string, npData *NodePortData) error {
```

Comment:
With cache indexer, this could be not needed.

Reply:
Updated.